### PR TITLE
Nk departments

### PR DESCRIPTION
**Pull request by nashvegasnate**
Full CRUD on Departments:
- A user can view the departments
- A user can create a department
- A user can edit a department
- A user can delete a department

Added some styling to home view, i.e., some product-type images that change every few seconds

### CODE REVIEW

File: client/src/components/NavBar.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const NavBar = ({ user, registeredUser }) => {
         <Link className="nav-link" to="/items">Items</Link>
       </NavItem>
       <NavItem>
+        <Link className="nav-link" to="/departments">Departments</Link>
```
Comment (caseywalker) [inline client/src/components/NavBar.js:29]: This needs to be on it's own Nav Item

File: client/src/components/EditDepartmentForm.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  departmentDescription,
+  departmentID
+}) {
+  const [department, setDepartment] = useState({
+    departmentName: departmentName || '',
+    departmentDescription: departmentDescription || '',
+    departmentID: departmentID || null,
+  });
+
+  const history = useHistory();
```
Comment (caseywalker) [inline client/src/components/EditDepartmentForm.js:26]: Are you using useHistory?

File: client/src/components/AddDepartmentForm.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  departmentName,
+  departmentDescription,
+  setShowButton,
+}) {
+  const [department, setDepartment] = useState({
+    departmentName: departmentName || '',
+    departmentDescription: departmentDescription || '',
+  });
+
+  const history = useHistory();
```
Comment (caseywalker) [inline client/src/components/AddDepartmentForm.js:25]: Are you using useHistory?

File: client/src/components/DepartmentCard.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+function DepartmentCard({
+  departmentID,
+  departmentName,
+  departmentDescription,
+  setDepartments,
+  user,
+}) {
+  const [updating, setUpdating] = useState(false);
+  const history = useHistory();
```
Comment (caseywalker) [inline client/src/components/DepartmentCard.js:23]: Are you using useHistory?

File: client/src/views/DepartmentsView.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+function Departments({ user }) {
+  const [departments, setDepartments] = useState([]);
+  const [showButton, setShowButton] = useState(false);
+  const handleClick = () => {
+    setShowButton((prevState) => !prevState);
+  };
+
+  useEffect(() => {
+    getDepartments(user.uid).then(setDepartments);
```
Comment (caseywalker) [inline client/src/views/DepartmentsView.js:28]: I don't think you need to pass user.uid to getDepartments

**Review by caseywalker — CHANGES_REQUESTED** [review]
Added comments that need to be resolved

**Review by fimoefive — COMMENTED** [review]
you only need useHistory() hook if you are doing the single card view, which I don't see that you are.

**Pull request closed without merging**